funcx/endpoint/endpoint.py

```diff
@@ -47,88 +71,70 @@ def check_pidfile(filepath, match_name, endpoint_name):
         logger.info('''Please cleanup using:
     $ funcx-endpoint stop {}'''.format(endpoint_name))
 
-def load_endpoint(endpoint_dir):
-    """
-    Parameters
-    ----------
-
-    endpoint_dir : str
-        endpoint directory path within funcx_dir
-    """
-    import importlib.machinery
-
-    endpoint_config_file = endpoint_dir + 'config.py'
-    endpoint_name = os.path.basename(endpoint_dir)
 
-    if os.path.exists(endpoint_config_file):
-        config = importlib.machinery.SourceFileLoader('{}_config'.format(endpoint_name),
-                                                      endpoint_config_file).load_module()
-    logger.debug("Loaded config for {}".format(endpoint_name))
-    return config.config
-
-def init_endpoint_dir(funcx_dir, endpoint_name, config_file=None):
+def init_endpoint_dir(endpoint_name, endpoint_config=None):
     """ Initialize a clean endpoint dir
 
     Returns if an endpoint_dir already exists
 
     Parameters
     ----------
-
-    funcx_dir : str
-        Path to the funcx_dir on the system
-
     endpoint_name : str
         Name of the endpoint, which will be used to name the dir
-        for the endpoint.
+        for the endpoint in the FUNCX_DIR
 
-    config_file : str
+    endpoint_config : str
        Path to a config file to be used instead of the funcX default config file
 
     """
-    endpoint_dir = os.path.join(funcx_dir, endpoint_name)
+    endpoint_dir = os.path.join(State.FUNCX_DIR, endpoint_name)
+    logger.debug(f"Creating endpoint dir {endpoint_dir}")
     os.makedirs(endpoint_dir, exist_ok=True)
 
-    if not config_file:
-        config_file = default_config.__file__
-
-    shutil.copyfile(config_file,
-                    os.path.join(endpoint_dir, 'config.py'))
+    endpoint_config = endpoint_config if endpoint_config else endpoint_default_config.__file__
+    shutil.copyfile(endpoint_config, os.path.join(endpoint_dir, FUNCX_CONFIG_FILE_NAME))
     return endpoint_dir
 
 
-def init_endpoint(args):
-    """ Setup funcx dirs and config files including a default endpoint config
+def init_endpoint():
+    """Setup funcx dirs and default endpoint config files
 
     TODO : Every mechanism that will update the config file, must be using a
     locking mechanism, ideally something like fcntl https://docs.python.org/3/library/fcntl.html
     to ensure that multiple endpoint invocations do not mangle the funcx config files
     or the lockfile module.
     """
-    funcx_dir = args.config_dir
-
-    if args.force and os.path.exists(funcx_dir):
-        logger.warning("Wiping all current configs in {}".format(funcx_dir))
+    _ = FuncXClient()
+ 
+    if os.path.exists(State.FUNCX_CONFIG_FILE):
```

Comment:
I don't think init_endpoint will ever be called if the funcx_config_file exists since `funcx-endpoint init` was removed. Good idea to ask for confirmation.